### PR TITLE
ADCM-2728 Rework upgrade function for starting task if upgrade has pending action

### PR DESCRIPTION
**Pull request by SofiaTal**

### CODE REVIEW

File: python/cm/api.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from django.db import transaction
 from django.utils import timezone
 
+import cm.upgrade
```
Comment (a-alferov) [inline python/cm/api.py:20]: Why?

File: python/cm/upgrade.py [hunk 2]
```diff
@@ -304,7 +305,7 @@ def rpm_cmp_reverse(obj1, obj2):
         return res
 
 
-def do_upgrade(obj: Union[Cluster, HostProvider], upgrade: Upgrade) -> dict:
+def do_upgrade(obj: Union[Cluster, HostProvider], upgrade: Upgrade):
```
Comment (a-alferov) [inline python/cm/upgrade.py:308]:
```suggestion
def do_upgrade(obj: Union[Cluster, HostProvider], upgrade: Upgrade) -> dict:
```

File: python/cm/upgrade.py [hunk 3]
```diff
@@ -313,9 +314,19 @@ def do_upgrade(obj: Union[Cluster, HostProvider], upgrade: Upgrade) -> dict:
...
-    if obj.prototype.type == 'cluster':
+    if not upgrade.action:
+        bundle_switch(obj, upgrade)
+    else:
+        cm.job.start_task(upgrade.action, obj, {}, {}, [], [], False)
+    obj.refresh_from_db()
+    return {'id': obj.id, 'upgradable': bool(get_upgrade(obj))}
+
+
+def bundle_switch(obj: Union[Cluster, HostProvider], upgrade: Upgrade) -> dict:
```
Comment (a-alferov) [inline python/cm/upgrade.py:325]:
```suggestion
def bundle_switch(obj: Union[Cluster, HostProvider], upgrade: Upgrade):
```

Comment (a-alferov) [reply python/cm/upgrade.py:325]: no return needed in this function

**Review by a-alferov — CHANGES_REQUESTED** [review]

File: python/cm/api.py (hunk 1 above)
Comment (SofiaTal) [reply python/cm/api.py:20]: I've tryed to fight with cyclic import in linters, but lose :(

**Review by a-alferov — APPROVED** [review]

**Pull request merged**